core/domain/exp_jobs_one_off.py

```diff
@@ -252,6 +252,60 @@ def reduce(key, values):
         yield (key, values)
 
 
+class ExplorationMigrationAuditJob(jobs.BaseMapReduceOneOffJobManager):
+    """A reusbale one-off job for testing exploration migration from the
+    previous exploration schema version to the latest. This job runs the state
+    migration, but does not commit the new exploration to the store.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @classmethod
+    def enqueue(cls, job_id, additional_job_params=None):
+        super(ExplorationMigrationAuditJob, cls).enqueue(
+            job_id, shard_count=64)
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+
+        current_state_schema_version = feconf.CURRENT_STATE_SCHEMA_VERSION
+        if item.states_schema_version == current_state_schema_version - 1:
+            try:
+                current_exp_schema_version = (
+                    exp_domain.Exploration.CURRENT_EXP_SCHEMA_VERSION)
+                conversion_fn = getattr(
+                    exp_domain.Exploration,
+                    '_convert_v%i_dict_to_v%i_dict' % (
+                        current_exp_schema_version - 1,
+                        current_exp_schema_version)
+                )
+                conversion_fn(item.to_dict())
+                yield ('SUCCESS', None)
+            except Exception as e:
+                error_message = (
+                    'Exploration %s failed migration to v%s: %s' %
+                    (item.id, current_exp_schema_version, e))
+                logging.error(error_message)
```

Comment:
Use logging.exception, since that would display the stacktrace as well.

Reply:
Done